mlir/lib/Target/Cpp/TranslateToCpp.cpp

```diff
@@ -1630,6 +1621,11 @@ LogicalResult CppEmitter::emitOperation(Operation &op, bool trailingSemicolon) {
        shouldBeInlined(cast<emitc::ExpressionOp>(op))))
     return success();
 
+  if (isa<cf::CondBranchOp, emitc::DeclareFuncOp, emitc::ForOp, emitc::IfOp,
+          emitc::VerbatimOp>(op)) {
+    trailingSemicolon = false;
```

Comment:
Maybe we can add a comment on the function that states that the flag is only used for operations where the semicolon is optional, e.g. expressions. But feel free to ignore it, since it is currently already a bit confusing when the flag is ignored or used.